### PR TITLE
Fix threads reply count sometimes off by one

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/element-web/issues/21415

---
This PR currently has no changelog labels, so will not be included in changelogs.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.

### CODE REVIEW

File: src/models/thread.ts [hunk 1]
```diff
@@ -191,7 +191,10 @@ export class Thread extends TypedEventEmitter<EmittedEvents, EventHandlerMap> {
         // There is a risk that the `localTimestamp` approximation will not be accurate
         // when threads are used over federation. That could results in the reply
         // count value drifting away from the value returned by the server
-        if (!this.lastEvent || (isThreadReply && event.localTimestamp > this.lastEvent.localTimestamp)) {
+        if (!this.lastEvent || (isThreadReply
+            && (event.getId() !== this.lastEvent.getId())
```
Comment (turt2live) [inline src/models/thread.ts:195]: would be good to get a comment here to say why we are checking the IDs (it implies there is code somewhere that is dealing with the `lastEvent`'s count, but not sure)

**Review by turt2live — APPROVED** [review]
lgtm either way

**Comment by turt2live** [comment]
/me notes to watch for auto-merge (it's not really a problem in this case)

**Comment by gsouquet** [comment]
@turt2live argh, sorry, when i enabled auto-merge I thought I had already had an approve and was waiting just for the CI.
Will try to address your comment as part of another PR

**Pull request merged**